frontend/packages/kubevirt-plugin/src/selectors/vm/selectors.ts

```diff
@@ -53,51 +54,14 @@ export const getVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[
 export const getDataVolumeTemplates = (vm: VMKind, defaultValue = []) =>
   _.get(vm, 'spec.dataVolumeTemplates') == null ? defaultValue : vm.spec.dataVolumeTemplates;
 
-export const getConfigMapVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('configMap'));
-
-export const getSecretVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('secret'));
-
-export const getServiceAccountVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] =>
-  getVolumes(vm, defaultValue).filter((vol) => Object.keys(vol).includes('serviceAccount'));
-
-export const getEnvDiskVolumes = (vm: VMKind, defaultValue: V1Volume[] = []): V1Volume[] => [
-  ...getConfigMapVolumes(vm, defaultValue),
-  ...getSecretVolumes(vm, defaultValue),
-  ...getServiceAccountVolumes(vm, defaultValue),
-];
-
-export const getConfigMapDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getConfigMapVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getSecretDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getSecretVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getServiceAccountDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !!getServiceAccountVolumes(vm).find((vol) => vol.name === disk.name),
-  );
-
-export const getEnvDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] => [
-  ...getConfigMapDisks(vm, defaultValue),
-  ...getSecretDisks(vm, defaultValue),
-  ...getServiceAccountDisks(vm, defaultValue),
-];
-
-export const getBootableDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter(
-    (disk) => !getEnvDisks(vm).find((envDisk) => envDisk.name === disk.name),
-  );
-
-export const getNonBootableDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] =>
-  getDisks(vm, defaultValue).filter((disk) =>
-    getEnvDisks(vm).find((envDisk) => envDisk.name === disk.name),
-  );
+export const getBootableDisks = (vm: VMKind, defaultValue: V1Disk[] = []): V1Disk[] => {
+  const volumeLookup = createBasicLookup(getVolumes(vm), getSimpleName);
+  return getDisks(vm, defaultValue).filter((disk) => {
+    const vol = volumeLookup[disk.name];
+    const volWrapper = new VolumeWrapper(vol);
```

Comment:
I think I saw this use case somewhere used as well. 

I suppose we could create a method on a wrapper called isEmpty (implemention by lodash isEmpty).
this way we would not have to check old stuff like `vol` - which could be invalidated (not here - in general) an just use a single variable volWrapper..

Reply:
although having a type specifies non empty

Reply:
my concern is referencing two variables in this line, when only one is used :)